SU2_CFD/include/variables/CVariable.hpp

```diff
@@ -93,7 +93,11 @@ class CVariable {
   MatrixType External_Old;   /*!< \brief Old external (outer) contribution in discrete adjoint multizone problems. */
 
   su2vector<bool> Non_Physical;  /*!< \brief Non-physical points in the solution (force first order). */
-
+  su2vector<unsigned short> Non_Physical_Counter; /*!< \brief Number of consecutive iterations that a point has been treated first-order. After a specified number of successful reconstructions, the point can be returned to second-order. */
+  
+  VectorType UnderRelaxation;  /*!< \brief Value of the under-relxation parameter local to the control volume. */
+  VectorType LocalCFL;         /*!< \brief Value of the CFL number local to the control volume. */
```

Comment:
I did not try that.. I am not sure if much can be gained since the equations are coupled and (most often) limited by the convective spectral radius (u + c). However, could be interesting to look into in the future. It would also require separate under-relaxation parameters for each equation too in order to drive the independent CFLs